config/config.go

```diff
@@ -172,6 +170,23 @@ func enrichFromEnvvars(config *HorizonConfig) error {
 		config.Edge.ExchangeMessageDynamicPoll = false
 	}
 
+	if apiPort := os.Getenv(AnaxAPIPort); apiPort != "" {
+		if config.Edge.APIListen != "" {
+			listen := strings.Split(config.Edge.APIListen, ":")
+			if len(listen) == 2 {
+				config.Edge.APIListen = fmt.Sprintf("%v:%v", listen[0], apiPort)
+			} else {
+				config.Edge.APIListen = fmt.Sprintf("127.0.0.1:%v", apiPort)
```

Comment:
When len(listen) == 1, should it also be `fmt.Sprintf("%v:%v", listen[0], apiPort)`?

Reply:
I am making the assumption that if len(listen) !=2 then it's an error because the format of the field is host:port, therefore len(listen)==1 means they only specified the host but not the port, that's why I set the host to loopback. How strongly do you feel about which way this should go?

Reply:
I can go either way. Since you allow overwrite the port in the /etc/default/horizon, I was thinking that user could just put the ip in the APIListen and let the env variable decide the port number.

Reply:
ok, that would be a change to the currently defined format of the field. I need to think about it again.